monitoring/configure-metrics-tools.sh

```diff
@@ -0,0 +1,94 @@
+wds="wds1"
+
+if [ "$1" == "--space-name" ]; then
+    ctx="$2"
+    shift 2
+fi
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
+
+# set core context
+kubectl config use-context $ctx
+
+: --------------------------------------------------------------------
+: Configure kubestellar controller for prometheus scraping
+: --------------------------------------------------------------------
+: 1. configure the rbac for the kube-rbac-proxy 
+sed s/%WDS_NS%/$wds-system/g kube-rbac-proxy-config.yaml | kubectl apply -f -
+
+: 2. Create svc for KS ctl manager
+kubectl -n $wds-system apply -f ${SCRIPT_DIR}/ks-ctl-manager-svc.yaml
+
+: 3. Create service monitor for KS ctl manager
+sed s/%WDS_NS%/$wds-system/g sm-ks-controller-manager.yaml | kubectl -n ks-monitoring apply -f -
+
+
+: --------------------------------------------------------------------
+: Configure kubestellar controller for pyroscope scraping
+: --------------------------------------------------------------------
+: 1. Open the pprof port:
+kubectl -n $wds-system get deploy kubestellar-controller-manager -o yaml | yq '(del(.status) |.spec.template.spec.containers.[1].ports[0].name |= "http")' | yq '.spec.template.spec.containers.[1].ports[0].protocol |= "TCP"' | yq '.spec.template.spec.containers.[1].ports[0].containerPort |= 8082' | kubectl --context $ctx apply --namespace=$wds-system -f -
+
+: 2. Add labels for pyroscope scraping:
+kubectl -n $wds-system get deploy kubestellar-controller-manager -o yaml | yq '.spec.template.metadata.annotations."profiles.grafana.com/cpu.port" |= "8082" |  .spec.template.metadata.annotations."profiles.grafana.com/cpu.scrape"|= "true" | .spec.template.metadata.annotations."profiles.grafana.com/goroutine.port" |= "8082" | .spec.template.metadata.annotations."profiles.grafana.com/goroutine.scrape" |= "true" |
+.spec.template.metadata.annotations."profiles.grafana.com/memory.port" |= "8082" | .spec.template.metadata.annotations."profiles.grafana.com/memory.scrape" |= "true"' | kubectl -n $wds-system apply -f -
+
+
+: --------------------------------------------------------------------
+: Configure transport controller for prometheus scraping
+: --------------------------------------------------------------------
+: 1. Create transport controller service:
+kubectl -n $wds-system apply -f ${SCRIPT_DIR}/ks-transport-ctl-svc.yaml
+
+: 2. Open the metrics and pprof ports:
+kubectl -n $wds-system get deploy transport-controller -o yaml | yq '(del(.status) |.spec.template.spec.containers.[0].ports[0].name |= "http1")' | yq '.spec.template.spec.containers.[0].ports[0].protocol |= "TCP"' | yq '.spec.template.spec.containers.[0].ports[0].containerPort |= 8090' | yq '.spec.template.spec.containers.[0].ports[1].name |= "http2"' | yq '.spec.template.spec.containers.[0].ports[1].protocol |= "TCP"' | yq '.spec.template.spec.containers.[0].ports[1].containerPort |= 8092' | kubectl --context $ctx apply --namespace=$wds-system -f -
+
```

Comment:
Why not configure for Pyroscope scraping?

Reply:
We are already configuring the transport controller for  Pyroscope scraping